contracts/pallet-ibc/src/ics20_fee/mod.rs

```diff
@@ -105,6 +105,21 @@ pub trait FlatFeeConverter {
 	) -> Option<u128>;
 }
 
+pub struct NonFlatFeeConverter<T>(PhantomData<T>);
```

Comment:
What’s the advantage over an enum that contains both cases?

Reply:
There is implementation of FlatFeeConverter in pablo-pallet
https://github.com/ComposableFi/composable/blob/main/code/parachain/frame/pablo/src/lib.rs#L667-L704

but composable-runtime does not depends on on pablo-pallet(that potentially can have enum implementation with both cases).

This NonFlatFeeConverter is a simple mock that can be used by any runtime that depends only from pallet-ibc and only from pallet-ibc.